pkgs/by-name/op/open-adventure/package.nix

```
@@ -0,0 +1,56 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitLab,
+  python312,
+  python312Packages,
+  pkg-config,
+  libedit,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "open-adventure";
+  version = "1.20";
+  src = fetchFromGitLab {
+    owner = "esr";
+    repo = "open-adventure";
+    rev = "refs/tags/${finalAttrs.version}";
+    hash = "sha256-xbsMz99CNLhpM6BSJVcRzxPB6tUYfPy/3Z+8BKt8b1E=";
+  };
+
+  nativeBuildInputs = [
+    python312
+    python312Packages.pyyaml
+    pkg-config
+  ];
+
+  buildInputs = [ libedit ];
+
+  buildPhase = ''
+    patchShebangs make_dungeon.py
+    make
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin
+    cp ./advent $out/bin
+    runHook postInstall
+  '';
+
+  meta = with lib; {
```

```suggestion
  meta = {
```
Nested `with;` expressions are considered a bad practice.